### PR TITLE
vpc_dev branch rebase to main branch

### PR DESCRIPTION
**Pull request by timdengyun**
This PR is to rebase vpc_dev to main branch in order to merge back all VPC dev PRs to main branch.
Hereafter, we will use main branch as our dev branch.

### CODE REVIEW

File: pkg/apis/nsx.vmware.com/v1alpha1/nsxserviceaccount_types.go [hunk 1]
```diff
@@ -0,0 +1,99 @@
+/* Copyright © 2022 VMware, Inc. All Rights Reserved.
```
Comment (gran-vmv) [inline pkg/apis/nsx.vmware.com/v1alpha1/nsxserviceaccount_types.go:1]: Why we have these dup files?

Comment (timdengyun) [reply pkg/apis/nsx.vmware.com/v1alpha1/nsxserviceaccount_types.go:1]: You mean file pkg/apis/nsx.vmware.com/v1alpha1/nsxserviceaccount_types.go and pkg/apis/v1alpha1/nsxserviceaccount_types.go? These two files are duplicated?

Comment (gran-vmv) [reply pkg/apis/nsx.vmware.com/v1alpha1/nsxserviceaccount_types.go:1]: Yes. Maybe whole `pkg/apis/nsx.vmware.com` is duplicated?

Comment (timdengyun) [reply pkg/apis/nsx.vmware.com/v1alpha1/nsxserviceaccount_types.go:1]: As I know, these files in pkg/apis/nsx.vmware.com folder is used when nsx operator is used as a module  @zhengxiexie please confirm this?

Comment (zhengxiexie) [reply pkg/apis/nsx.vmware.com/v1alpha1/nsxserviceaccount_types.go:1]: Yes

Comment (gran-vmv) [reply pkg/apis/nsx.vmware.com/v1alpha1/nsxserviceaccount_types.go:1]: Any method to make these 2 folder synchronized, or at least notify developers to sync it manually?

**Review by seanpang-vmware — DISMISSED** [review]
VPC part LGTM

File: pkg/apis/nsx.vmware.com/v1alpha1/nsxserviceaccount_types.go (hunk 1 above)
Comment (zhengxiexie) [reply pkg/apis/nsx.vmware.com/v1alpha1/nsxserviceaccount_types.go:1]: `make generated` already sync

**Comment by zhengxiexie** [comment]
pls add https://github.com/vmware-tanzu/nsx-operator/pull/401

**Comment by timdengyun** [comment]
> pls add #401

https://github.com/vmware-tanzu/nsx-operator/pull/399/commits/57201b6268814a44f8318aa126dce89f63ee51ce
Pick this CLN to this PR. please take a look.

File: pkg/apis/nsx.vmware.com/v1alpha1/nsxserviceaccount_types.go (hunk 1 above)
Comment (timdengyun) [reply pkg/apis/nsx.vmware.com/v1alpha1/nsxserviceaccount_types.go:1]: @zhengxiexie  I found the files in pkg/apis/nsx.vmware.com/v1alpha1/2 has some differences with the files in pkg/apis/v1alpha1/2, Could you help check which one is latest version?  if necessary, we might need to a new PR to sync these two folders.

**Comment by zhengxiexie** [comment]
LGTM

**Comment by zhengxiexie** [comment]
> > pls add #401
> > @zhengxiexie
> > [57201b6](https://github.com/vmware-tanzu/nsx-operator/commit/57201b6268814a44f8318aa126dce89f63ee51ce)
> > Pick this CLN to this PR. please take a look.

also 
https://github.com/vmware-tanzu/nsx-operator/pull/403
https://github.com/vmware-tanzu/nsx-operator/pull/404

File: pkg/apis/nsx.vmware.com/v1alpha1/nsxserviceaccount_types.go (hunk 1 above)
Comment (timdengyun) [reply pkg/apis/nsx.vmware.com/v1alpha1/nsxserviceaccount_types.go:1]: @gran-vmv  https://github.com/vmware-tanzu/nsx-operator/pull/402 yes, this PR is to sync files between two places.  These files are synced by make generated

**Review by seanpang-vmware — APPROVED** [review]
LGTM

**Pull request merged**